src/rgw/rgw_common.h

```diff
@@ -1114,6 +1115,11 @@ struct RGWBucketInfo {
   // layout of bucket index objects
   rgw::BucketLayout layout;
 
+  // layout of bucket index logs
+  rgw::bucket_log_layout_generation current_log_layout;
+
+  std::vector<rgw::bucket_log_layout_generation> log_layouts;
+
```

Comment:
please put these new fields inside of `rgw::BucketLayout`, so that they're kept private to the zone too

we can also move all of the new types from rgw_bucket_log_layout.h into rgw_bucket_layout.h